# Questions about Beta

## *What is Beta?*

- A measure of a stock's volatility relative to the overall market

## *If a stock has a Beta value of 1.2, what does that mean?*

- A beta value of 1.2 implies that a stock is 20% more volatile than the market overall
    - E.g. if the market goes up by 10%, we'd expect the stock to go up by 12%

## *Can Beta be negative?*

- Yes
    - If a stock moves in the opposite direction as the market, its Beta value will be negative
        - **Example**: Gold - as the market goes down, investors sell stocks and buy gold, giving gold a negative Beta value

## *What is the formula for calculating Beta?*

$$
\beta = \frac{Cov(R_{Stock}, R_{Market})}{Var(R_{Market})}
$$

- $R_{Stock}$ is the return for the stock we're analyzing
- $R_{Market}$ is the return for the market overall
    - Often, we use the S&P 500 to represent the market
- So, $Cov(R_{Stock}, R_{Market})$ represents the covariance of the returns
- $Var(R_{Market})$ is the variance of the market's returns

## *What is the Beta of AMZN?*

- We'll calculate this using the yahoo finance package

In [1]:
import datetime

import numpy as np
import pandas as pd
import yfinance as yf
from dateutil.relativedelta import relativedelta

In [2]:
sp500_ticker = "^GSPC"  # S&P 500 index
amzn_ticker = "AMZN"  # Amazon stock

- We'll get the data going back 5 years

In [3]:
today = datetime.datetime.today().date()
end = today.strftime("%Y-%m-%d")
offset_5_years = relativedelta(years=5)
start = (today - offset_5_years).strftime("%Y-%m-%d")
start, end

('2019-12-11', '2024-12-11')

In [4]:
df_all = yf.download([sp500_ticker, amzn_ticker], start=start, end=end)
df_all

[*********************100%***********************]  2 of 2 completed


Price                       Adj Close                    Close               \
Ticker                           AMZN        ^GSPC        AMZN        ^GSPC   
Date                                                                          
2019-12-11 00:00:00+00:00   87.435997  3141.629883   87.435997  3141.629883   
2019-12-12 00:00:00+00:00   88.016502  3168.570068   88.016502  3168.570068   
2019-12-13 00:00:00+00:00   88.046997  3168.800049   88.046997  3168.800049   
2019-12-16 00:00:00+00:00   88.460503  3191.449951   88.460503  3191.449951   
2019-12-17 00:00:00+00:00   89.532997  3192.520020   89.532997  3192.520020   
...                               ...          ...         ...          ...   
2024-12-04 00:00:00+00:00  218.160004  6086.490234  218.160004  6086.490234   
2024-12-05 00:00:00+00:00  220.550003  6075.109863  220.550003  6075.109863   
2024-12-06 00:00:00+00:00  227.029999  6090.270020  227.029999  6090.270020   
2024-12-09 00:00:00+00:00  226.089996  6052.850098  226.089996  6052.850098   
2024-12-10 00:00:00+00:00  225.039993  6034.910156  225.039993  6034.910156   

Price                            High                      Low               \
Ticker                           AMZN        ^GSPC        AMZN        ^GSPC   
Date                                                                          
2019-12-11 00:00:00+00:00   87.500000  3143.979980   86.785500  3133.209961   
2019-12-12 00:00:00+00:00   88.199997  3176.280029   87.272003  3138.469971   
2019-12-13 00:00:00+00:00   88.449501  3182.679932   87.750000  3156.510010   
2019-12-16 00:00:00+00:00   88.474998  3197.709961   87.852501  3183.629883   
2019-12-17 00:00:00+00:00   89.599998  3198.219971   88.869499  3191.030029   
...                               ...          ...         ...          ...   
2024-12-04 00:00:00+00:00  220.000000  6089.839844  215.750000  6061.060059   
2024-12-05 00:00:00+00:00  222.149994  6094.549805  217.300003  6072.899902   
2024-12-06 00:00:00+00:00  227.149994  6099.970215  220.600006  6079.979980   
2024-12-09 00:00:00+00:00  230.080002  6088.509766  225.669998  6048.629883   
2024-12-10 00:00:00+00:00  229.059998  6065.399902  224.199997  6029.890137   

Price                            Open                 Volume              
Ticker                           AMZN        ^GSPC      AMZN       ^GSPC  
Date                                                                      
2019-12-11 00:00:00+00:00   87.083504  3135.750000  41952000  3257650000  
2019-12-12 00:00:00+00:00   87.500000  3141.229980  61918000  4003200000  
2019-12-13 00:00:00+00:00   88.250000  3166.649902  54914000  3757650000  
2019-12-16 00:00:00+00:00   88.349998  3183.629883  62904000  4070200000  
2019-12-17 00:00:00+00:00   88.900497  3195.399902  72888000  3842940000  
...                               ...          ...       ...         ...  
2024-12-04 00:00:00+00:00  215.960007  6069.390137  48745700  4003390000  
2024-12-05 00:00:00+00:00  218.029999  6089.029785  41140200  4212020000  
2024-12-06 00:00:00+00:00  220.750000  6081.379883  44178100  3924830000  
2024-12-09 00:00:00+00:00  227.210007  6083.009766  46819400  4556460000  
2024-12-10 00:00:00+00:00  226.089996  6057.589844  31145600  4048410000  

[1258 rows x 12 columns]

- We'll only use the `Adj Close` column

In [5]:
df_daily_returns = df_all["Adj Close"].pct_change().dropna()
df_daily_returns

Ticker                         AMZN     ^GSPC
Date                                         
2019-12-12 00:00:00+00:00  0.006639  0.008575
2019-12-13 00:00:00+00:00  0.000346  0.000073
2019-12-16 00:00:00+00:00  0.004696  0.007148
2019-12-17 00:00:00+00:00  0.012124  0.000335
2019-12-18 00:00:00+00:00 -0.003703 -0.000432
...                             ...       ...
2024-12-04 00:00:00+00:00  0.022114  0.006051
2024-12-05 00:00:00+00:00  0.010955 -0.001870
2024-12-06 00:00:00+00:00  0.029381  0.002495
2024-12-09 00:00:00+00:00 -0.004140 -0.006144
2024-12-10 00:00:00+00:00 -0.004644 -0.002964

[1257 rows x 2 columns]

- Calculating the covariance matrix

In [6]:
covariance_matrix = np.cov(df_daily_returns["AMZN"], df_daily_returns["^GSPC"])
covariance_matrix

array([[0.00051211, 0.00019396],
       [0.00019396, 0.00017967]])

#### *What do these values mean?*

- The covariance matrix represents:

$$
\text{Cov} =
\begin{bmatrix}
\text{Var}(\text{AMZN}) & \text{Cov}(\text{AMZN}, \text{GSPC}) \\
\text{Cov}(\text{AMZN}, \text{GSPC}) & \text{Var}(\text{GSPC})
\end{bmatrix}
$$

- So, the value we want is in the top right
    - **Recall**: since covariance is symmetric (i.e. $Cov(A,B) = Cov(B,A)$), we could also take the value from the bottom left

In [7]:
covariance = covariance_matrix[0, 1]
covariance

0.00019395785734819564

- Next, we need the variance of the S&P 500
    - This is already calculated in the covariance matrix as well

In [8]:
variance = covariance_matrix[1, 1]
variance

0.00017967009904843253

- Now, we have everything we need to calculate Beta:

In [9]:
beta = covariance / variance
beta

1.0795221819069163

- Therefore, we calculated a Beta value of:

$$
Beta_{AMZN} = \frac{Cov(R_{AMZN}, R_{\text{S\&P 500}})}{Var(R_{\text{S\&P 500}})} = \frac{0.00019395785734819564}{0.00017967009904843253} = 1.0795
$$

#### *What if we used monthly returns instead?*

In [10]:
df_monthly_returns = df_all["Adj Close"].resample("ME").last().pct_change().dropna()
df_monthly_returns

Ticker                         AMZN     ^GSPC
Date                                         
2020-01-31 00:00:00+00:00  0.087064 -0.001628
2020-02-29 00:00:00+00:00 -0.062214 -0.084110
2020-03-31 00:00:00+00:00  0.035021 -0.125119
2020-04-30 00:00:00+00:00  0.268900  0.126844
2020-05-31 00:00:00+00:00 -0.012785  0.045282
2020-06-30 00:00:00+00:00  0.129567  0.018388
2020-07-31 00:00:00+00:00  0.147114  0.055101
2020-08-31 00:00:00+00:00  0.090461  0.070065
2020-09-30 00:00:00+00:00 -0.087579 -0.039228
2020-10-31 00:00:00+00:00 -0.035754 -0.027666
2020-11-30 00:00:00+00:00  0.043440  0.107546
2020-12-31 00:00:00+00:00  0.028058  0.037121
2021-01-31 00:00:00+00:00 -0.015576 -0.011137
2021-02-28 00:00:00+00:00 -0.035328  0.026091
2021-03-31 00:00:00+00:00  0.000372  0.042439
2021-04-30 00:00:00+00:00  0.120663  0.052425
2021-05-31 00:00:00+00:00 -0.070470  0.005486
2021-06-30 00:00:00+00:00  0.067355  0.022214
2021-07-31 00:00:00+00:00 -0.032722  0.022748
2021-08-31 00:00:00+00:00  0.043034  0.028990
2021-09-30 00:00:00+00:00 -0.053518 -0.047569
2021-10-31 00:00:00+00:00  0.026602  0.069144
2021-11-30 00:00:00+00:00  0.039924 -0.008334
2021-12-31 00:00:00+00:00 -0.049252  0.043613
2022-01-31 00:00:00+00:00 -0.102830 -0.052585
2022-02-28 00:00:00+00:00  0.026673 -0.031360
2022-03-31 00:00:00+00:00  0.061437  0.035773
2022-04-30 00:00:00+00:00 -0.237525 -0.087957
2022-05-31 00:00:00+00:00 -0.032764  0.000053
2022-06-30 00:00:00+00:00 -0.116459 -0.083920
2022-07-31 00:00:00+00:00  0.270596  0.091116
2022-08-31 00:00:00+00:00 -0.060615 -0.042440
2022-09-30 00:00:00+00:00 -0.108622 -0.093396
2022-10-31 00:00:00+00:00 -0.093451  0.079863
2022-11-30 00:00:00+00:00 -0.057595  0.053753
2022-12-31 00:00:00+00:00 -0.129894 -0.058971
2023-01-31 00:00:00+00:00  0.227738  0.061753
2023-02-28 00:00:00+00:00 -0.086299 -0.026112
2023-03-31 00:00:00+00:00  0.096148  0.035052
2023-04-30 00:00:00+00:00  0.020912  0.014642
2023-05-31 00:00:00+00:00  0.143480  0.002482
2023-06-30 00:00:00+00:00  0.081108  0.064727
2023-07-31 00:00:00+00:00  0.025468  0.031139
2023-08-31 00:00:00+00:00  0.032391 -0.017716
2023-09-30 00:00:00+00:00 -0.078907 -0.048719
2023-10-31 00:00:00+00:00  0.046963 -0.021980
2023-11-30 00:00:00+00:00  0.097678  0.089179
2023-12-31 00:00:00+00:00  0.040044  0.044229
2024-01-31 00:00:00+00:00  0.021456  0.015896
2024-02-29 00:00:00+00:00  0.138918  0.051721
2024-03-31 00:00:00+00:00  0.020480  0.031019
2024-04-30 00:00:00+00:00 -0.029826 -0.041615
2024-05-31 00:00:00+00:00  0.008229  0.048021
2024-06-30 00:00:00+00:00  0.095273  0.034670
2024-07-31 00:00:00+00:00 -0.032445  0.011321
2024-08-31 00:00:00+00:00 -0.045352  0.022835
2024-09-30 00:00:00+00:00  0.043866  0.020197
2024-10-31 00:00:00+00:00  0.000376 -0.009897
2024-11-30 00:00:00+00:00  0.115290  0.057301
2024-12-31 00:00:00+00:00  0.082496  0.000419

In [11]:
covariance_matrix = np.cov(df_monthly_returns["AMZN"], df_monthly_returns["^GSPC"])
covariance = covariance_matrix[0, 1]
variance = covariance_matrix[1, 1]
beta = covariance / variance
beta

1.157850559913722

- This matches the result from a simple Google search
    - Nice

## *Why are the daily and monthly Beta values different?*

- Because they represent different things
    - Intuitively, we know a lot more can happen in a month than a day
        - Therefore, **we expect monthly returns to be more volatile**
            - We can see this in our example: the monthly Beta is higher than the daily Beta
- So, monthly Beta is better for predicting longer-term performance
    - Daily Beta is better for short-term trading decisions

____

# Summary

1. Beta measures the volatility of a stock relative to the market overall
2. Beta is the ratio of covariance to market variance